File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
   let hash={}

   for(let i=0;i<nums.length;i++)
   {
    if(target-nums[i] in hash)
    {
        return[hash[target-nums[i]],i]
    }
    hash[nums[i]]=i
   }

};

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
     vector<int>ans;
     unordered_map<int,int>mpp;

     for(int i=0;i<nums.size();i++)
     {
         if(mpp.find(target-nums[i])!=mpp.end())
         {//if the difference is found in the map return the current index and index of difference as pair 
             ans.push_back(mpp[target-nums[i]]);
             ans.push_back(i);
             return ans;
         }
         mpp[nums[i]]=i; //otherwise store the difference as key value in map
     }   
     return ans;
    }
};